src/utils.cpp

```diff
@@ -107,25 +108,21 @@ int get_word_end(const char* text, int start) {
 }
 
 std::optional<std::string> read_file_to_string(const char* path) {
-    FILE* f = fopen(path, "r");
-    if (!f) return std::nullopt;
+    std::ifstream input_stream {path, std::fstream::in | std::fstream::ate};
+    if (!input_stream) return std::nullopt;
 
-    fseek(f, 0, SEEK_END);
-    size_t size = ftell(f);
+    const std::size_t size = input_stream.tellg();
+    std::string buffer(size, '\0');
 
-    std::string contents;
-    contents.resize(size);
+    input_stream.seekg(0);
```

Comment:
Do we need to seek here?

Reply:
Since we use std::ios::ate (std::fstream::ate), we open the stream at the end of the file thus we need to move back to the start to read the contiguous buffer 😃